pandas/tests/arrays/categorical/test_api.py

```diff
@@ -337,12 +338,31 @@ def test_remove_categories(self):
         # inplace == True
         res = cat.remove_categories("c", inplace=True)
         tm.assert_categorical_equal(cat, new)
+
+        # for removing duplicates
+        cat = Categorical(["a", "b", "c", "a"], ordered=True)
+        old = cat.copy()
+        new = Categorical([np.nan, "b", "c", np.nan], categories=["b", "c"], ordered=True)
+
+        res = cat.remove_categories("a")
+        tm.assert_categorical_equal(cat, old)
+        tm.assert_categorical_equal(res, new)
+
+        res = cat.remove_categories(["a"])
+        tm.assert_categorical_equal(cat, old)
+        tm.assert_categorical_equal(res, new)
+
         assert res is None
 
         # removal is not in categories
-        with pytest.raises(ValueError):
+        msg = re.escape("removals must all be in old categories: ['c']")
+        with pytest.raises(ValueError, match=msg):
             cat.remove_categories(["c"])
 
+        msg = re.escape("removals must all be in old categories: [np.nan]")
+        with pytest.raises(ValueError):
+            cat.remove_categories([np.nan])
```

Comment:
I think the 4 test cases should be..

```
cat.remove_categories(["c"])
cat.remove_categories(["c", np.nan])
cat.remove_categories("c")
cat.remove_categories(["c","c" ])
```
 and all these cases should raise the same exception with the same message.

if you could split the pytest raises test from `test_remove_categories` into a new test `test_remove_categories_raises` and parameterize that would be great.

Reply:
Shall I do it for all the above stated ones?

Reply:
The first two are the ones in the OP #28669, so necessary to close issue.

The second two as suggested in https://github.com/pandas-dev/pandas/issues/28669#issuecomment-536309377

Reply:
I've the all the cases.